packages/property-grid/src/components/PresentationPropertyGridWidget2.tsx

```diff
@@ -0,0 +1,772 @@
+                  icon: "icon-add",
+                  onSelect: async () => onAddFavorite(field),
+                  title: localizations.addFavorite.title,
+                  label: localizations.addFavorite.label,
+                });
+              }
+            }
+
+            if (props.enableCopyingPropertyText) {
+              items.push({
+                key: "copy-text",
+                onSelect: async () => {
+                  if (props.featureTracking)
+                    props.featureTracking.trackCopyPropertyText();
+                  await onCopyText(args.propertyRecord);
+                },
+                title: localizations.copyText.title,
+                label: localizations.copyText.label,
+              });
+            }
+
+            if (props.enableNullValueToggle || true) {
+              if (showNullValues) {
+                items.push({
+                  key: "hide-null",
+                  onSelect: () => {
+                    onHideNull();
+                  },
+                  title: localizations.hideNull.title,
+                  label: localizations.hideNull.label,
+                });
+              } else {
+                items.push({
+                  key: "show-null",
+                  onSelect: () => {
+                    onShowNull();
+                  },
+                  title: localizations.showNull.title,
+                  label: localizations.showNull.label,
+                });
+              }
+            }
+            setContextMenu(args);
+            setContextMenuItemInfos(items.length > 0 ? items : undefined);
+          });
+      }
+    },
+    [
+      iModelConnection,
+      dataProvider,
+      localizations,
+      sharedFavorites,
+      showNullValues,
+      props,
+      onAddFavorite,
+      onRemoveFavorite,
+      onShareFavorite,
+      onUnshareFavorite,
+      onCopyText,
+      onHideNull,
+      onShowNull,
+    ]
+  );
+
+  const onPropertyContextMenu = React.useCallback(
+    (args: PropertyGridContextMenuArgs) => {
+      args.event.persist();
+      setupContextMenu(args);
+    },
+    [setupContextMenu]
+  );
+
+  const onContextMenuOutsideClick = React.useCallback(() => {
+    setContextMenu(undefined);
+  }, []);
+
+  const onContextMenuEsc = React.useCallback(() => {
+    setContextMenu(undefined);
+  }, []);
+
+  const favoriteActionButtonRenderer = React.useCallback(
+    (_props: ActionButtonRendererProps) => {
+      if (iModelConnection && dataProvider) {
+        const { property } = _props;
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        const field = useAsyncValue(
+          // eslint-disable-next-line react-hooks/rules-of-hooks
+          React.useMemo(
+            async () => dataProvider.getFieldByPropertyRecord(property),
+            [property]
+          )
+        );
+
+        return (
+          <div>
+            {field &&
+              (Presentation.favoriteProperties.has(
```

Comment:
same thing about that callback

Reply:
Similarly had a warning that suggested using useCallback